### PR TITLE
Change XXPlusYYGate convention

### PR DESCRIPTION
**Pull request by kevinsung**
### Summary
Changes the convention so that the phase matches with https://arxiv.org/pdf/1912.04424.pdf but the sign of the exponential is consistent with other Qiskit gates.

### Details and comments
This is a breaking change.

### CODE REVIEW

**Review by jakelishman — APPROVED** [review]
I don't know why this sat for so long when we'd decided on it in the meeting.  Sorry about that - merging now.

**Review by jakelishman — CHANGES_REQUESTED** [review]
No wait, sorry - please could you add an `upgrade` release note that says that the convention changed?

**Comment by kevinsung** [comment]
@jakelishman Done.

File: releasenotes/notes/xxplusyy-convention-e181e74271a9e9e1.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+upgrade:
+  - |
+    The definition of :class:`.XXPlusYYGate` has been changed.
+    See https://github.com/Qiskit/qiskit-terra/pull/7949
```
Comment (jakelishman) [inline releasenotes/notes/xxplusyy-convention-e181e74271a9e9e1.yaml:5]:
```suggestion
    See `#7969 <https://github.com/Qiskit/qiskit-terra/pull/7949>`__
```

**Review by jakelishman — APPROVED** [review]
Great, thanks!

**Pull request merged**